### PR TITLE
mark `segfault` function more reliable and return !

### LINKED ISSUE #4: `segfault::segfault()` should be marked as `-> !` since they do not return
I believe, this may make it easier for the compiler to optimise the segfault, otherwise, it is too slow for my purposes.

### PR DESCRIPTION
**Pull request by Oderjunkie**
fixes #4

### CODE REVIEW

**Review by Popaulol — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
-pub fn segfault() {
+pub fn segfault() -> ! {
     unsafe { std::ptr::null_mut::<i32>().write(1) }
+    unreachable!();
```
Comment (Vendicated) [inline src/lib.rs]: this crate takes 🚀 performance 🚀 very seriously, so `std::hint::unreachable_unchecked()` should be used instead to greatly 🚀 skyrocket 🚀 performance 🚀

**Comment by Oderjunkie** [comment]
also fixes #1 now

**Comment by Vendicated** [comment]
thank you so much for this great empowering PR!!

**Comment by Vendicated** [comment]
i released v0.1.1 to crates.io

**Pull request merged**